plugins/org.jkiss.dbeaver.ui/src/org/jkiss/dbeaver/ui/controls/CustomTimeEditor.java

```diff
@@ -271,10 +272,11 @@ public void setTextValue(@Nullable String value) {
         }
     }
 
-    public void setValue(@Nullable Date value) {
+    public void setValue(@Nullable Object value) throws ClassCastException {
         updateWarningLabel(value);
-        if (value != null) {
-            calendar.setTime(value);
+        Date adaptedValue = adaptToDate(value);
+        if (adaptedValue != null) {
+            calendar.setTime(adaptedValue);
         } else {
             switch (inputMode) {
```

Comment:
An off-topic nitpick: why are the enum `InputMode` constants named in CamelCase instead of SNAKE_CASE?

Reply:
Yep, this one looks like my old code mistake I will fix it.